### ISSUE TITLE
Feature Request: Automatic Label Generation

### ISSUE THREAD

**Issue by batonac** [ParsimonyGit/shipstation_integration#66]
I'm not sure how this would work yet, but I'm needing some way to automatically generate the labels for orders based on the order item(s) weigh and size.

I suppose there needs to be some kind of defaults table per store with the ability to assign carriers and services for things like order weight, category, customer class, etc.

Is this a scenario for any of your clients? Any ideas of where to get started?

**Comment by agritheory** [comment]
@batonac Yo! The label generation API has worked for me in the past in previous iterations of this app. Also, I think we want to manually generate labels because you are charged per label. Both approaches have room for abuse, but only one can be charged back to the developer. 

As far as setting defaults, there are _multiple_ ways to solve this but I think that leveraging/ extending the Shipping Rule doctype makes the most sense.

**Comment by Alchez** [comment]
@batonac, I agree with @agritheory in keeping label generation responsible to an actual user considering it's going to rack up costs. We haven't had a client who has wanted auto-generation of labels yet, but if you think it's something worth looking into, feel free to add a design spec here.
